Sources/TuistCore/Models/TestIdentifier.swift

```diff
@@ -0,0 +1,58 @@
+import TuistSupport
+
+public struct TestIdentifier: CustomStringConvertible, Hashable {
+    enum Error: FatalError {
+        case invalidTestIdentifier(value: String)
+
+        // Error description
+        var description: String {
+            switch self {
+            case let .invalidTestIdentifier(value):
+                return "Invalid test identifiers \(value). The expected format is TestTarget[/TestClass[/TestMethod]]."
+            }
+        }
+
+        // Error type
+        var type: ErrorType {
+            switch self {
+            case .invalidTestIdentifier:
+                return .abort
+            }
+        }
+    }
+
+    public let target: String
+    public let `class`: String?
```

Comment:
Maybe we should rename this to `className` , so we don't have to escape it everywhere?